src/mesh/logical_location.hpp

```diff
@@ -122,19 +154,20 @@ class LogicalLocation { // aggregate and POD type
   LogicalLocation
   GetSameLevelNeighbor(int ox1, int ox2, int ox3,
                        const RootGridInfo &rg_info = RootGridInfo()) const {
-    return LogicalLocation(level(), lx1() + ox1, lx2() + ox2, lx3() + ox3);
+    return LogicalLocation(tree(), level(), lx1() + ox1, lx2() + ox2, lx3() + ox3);
   }
 
-  LogicalLocation GetParent() const {
-    if (level() <= 0) return LogicalLocation(level() - 1, 0, 0, 0);
-    return LogicalLocation(level() - 1, lx1() >> 1, lx2() >> 1, lx3() >> 1);
+  LogicalLocation GetParent(int nlevel = 1) const {
```

Comment:
nlevel is the offset wrt to number of levels for the parent, i.e., nlevel=2 would give the grandparent?

Reply:
yep, that is right.